math/eigen/include/algebra/math/impl/eigen_transform3.hpp

```diff
@@ -128,6 +132,14 @@ struct transform3 {
     _data_inv = _data.inverse();
   }
 
+  /// Constructor with arguments: matrix and its inverse
+  ///
+  /// @param m is the full 4x4 matrix
+  /// @param m_inv is the inverse to m
+  ALGEBRA_HOST_DEVICE
+  transform3(const matrix44 &m, const matrix44 &m_inv)
+      : _data{m}, _data_inv{m_inv} {}
```

Comment:
Would it make sense to add a debug-mode assertion that `m_inv` is indeed the inverse if `m`?

Reply:
Done